core/src/language/swift.rs

```diff
@@ -261,51 +266,38 @@ impl Language for Swift {
         writeln!(w)?;
         self.write_comments(w, 0, &rs.comments)?;
 
-        let type_name = swift_keyword_aware_rename(&format!("{}{}", self.prefix, rs.id.renamed));
+        let type_name =
+            swift_keyword_aware_rename(Cow::Owned(format!("{}{}", self.prefix, rs.id.renamed)));
 
         // If there are no decorators found for this struct, still write `Codable` and default decorators for structs
-        let mut decs = self.get_default_decorators();
-
-        let default_generic_constraints = self.default_generic_constraints.clone();
         // Check if this struct's decorators contains swift in the hashmap
-        if let Some(swift_decs) = rs.decorators.get(&SupportedLanguage::Swift) {
+        let decs = if let Some(swift_decs) = rs.decorators.get(&DecoratorKind::Swift) {
             // For reach item in the received decorators in the typeshared struct add it to the original vector
             // this avoids duplicated of `Codable` without needing to `.sort()` then `.dedup()`
             // Note: the list received from `rs.decorators` is already deduped
-            swift_decs
-                .iter()
-                .filter(|d| d.as_str() != CODABLE)
-                .for_each(|d| decs.push(d.clone()));
+            Either::Left(
+                self.get_default_decorators().chain(
+                    swift_decs
+                        .iter()
+                        .filter(|d| d.as_str() != CODABLE)
+                        .map(|s| s.as_str()),
+                ),
+            )
+        } else {
+            Either::Right(self.get_default_decorators())
```

Comment:
This is a nice pattern! I don't think I've seen `Either` used like this for the two branches to allow a simple iterator fold at the end, but it's a lot more readable than collecting into an intermediate vector.